app/models/user_role.rb

```diff
@@ -124,6 +124,18 @@ def self.is_eligible_voter?(role)
     end
   end
 
+  def self.discourse_user_group(role)
+    is_actual_role = role.is_a?(UserRole) # Eventually, all roles will be migrated to the new system, till then some roles will actually be hashes.
+    group_type = is_actual_role ? role.group.group_type : role[:group][:group_type]
+    group = is_actual_role ? role.group : role[:group]
```

Comment:
Shouldn't this use the `UserGroup.group_type(foo)` helper methods?

Reply:
Oh yes, I think I copied this from somewhere and hence missed it. Changed it now, thanks.